Sources/SourceKitLSP/Swift/SwiftLanguageService.swift

```diff
@@ -741,9 +741,19 @@ extension SwiftLanguageService {
         """
     }
 
+    var tokenRange: Range<Position>?
+
+    if let snapshot = try? await latestSnapshot(for: uri) {
+      let tree = await syntaxTreeManager.syntaxTree(for: snapshot)
+      if let token = tree.token(at: snapshot.absolutePosition(of: position)) {
+        let range: Range<Position> = snapshot.absolutePositionRange(of: token.position..<token.endPosition)
+        tokenRange = range
```

Comment:
Couldn’t this just be the following?

```suggestion
        tokenRange = snapshot.absolutePositionRange(of: token.position..<token.endPosition)
```

Reply:
of course ! 
i think i was doing something more with the variable and forgot to clean it. 
done here 
https://github.com/swiftlang/sourcekit-lsp/pull/1899/commits/b7ee7308d758ced6847f6750bbcae7308839ea18